pkg/providers/tinkerbell/upgrade.go

```diff
@@ -113,8 +113,12 @@ func (p *Provider) SetupAndValidateUpgradeCluster(ctx context.Context, cluster *
 		return err
 	}
 
-	if err := p.validateAvailableHardwareForUpgrade(ctx, currentClusterSpec, clusterSpec); err != nil {
-		return err
+	upgradeStrategy := currentClusterSpec.Cluster.Spec.ControlPlaneConfiguration.UpgradeRolloutStrategy
```

Comment:
I guess this assumes that both CP/worker in place strategy will be the same, which aligns with how we currently have it configured. I know we discussed potentially changing that limitation, so adding a note to keep this validation in mind if we do change it.